js/_hole-common.tsx

```diff
@@ -197,8 +197,19 @@ export function setCode(code: string, editor: EditorView | null) {
 }
 
 function updateLangPicker() {
-    // Populate the language picker with accurate stroke counts.
-    $('#picker').replaceChildren(...sortedLangs.map((l: any) => {
+    const langSelect = $('#langSelect') as HTMLSelectElement;
+    if (langSelect.childElementCount === 0) {
+        for (const l of sortedLangs as any[]) {
+            const suffix = l.experiment ? ' (experimental)' : '';
+            langSelect.appendChild(<option value={l.id}>{l.name}{suffix}</option>);
+        }
+
+        langSelect.addEventListener('change', () => location.hash = '#' + langSelect.value);
+    }
+    langSelect.value = lang;
+
+    // Hybrid language selector: make it easy to see your existing solutions and their lengths.
+    $('#picker').replaceChildren(...sortedLangs.filter((l: any) => getSolutionCode(l.id, 0)).map((l: any) => {
```

Comment:
I don't like that we're calling `getSolutionCode` twice now, since we already do it in side the map, why not just return `null` inside the map in the else clause?

Reply:
I tried it and it didn't seem to work. It ended up saying nullnullnull... or something on the page. But if I just add a filter after the map that would work.